src/main/java/me/zeroeightsix/kami/mixin/client/player/MixinEntityPlayer.java

```diff
@@ -20,11 +24,14 @@ public MixinEntityPlayer(World worldIn) {
 
     @Inject(method = "travel", at = @At("HEAD"), cancellable = true)
     public void travel(float strafe, float vertical, float forward, CallbackInfo info) {
-        PlayerTravelEvent event = new PlayerTravelEvent();
-        KamiEventBus.INSTANCE.post(event);
-        if (event.getCancelled()) {
-            move(MoverType.SELF, motionX, motionY, motionZ);
-            info.cancel();
+        //noinspection ConstantConditions
+        if (EntityPlayerSP.class.isAssignableFrom(this.getClass())) {
```

Comment:
I hate that IDEs complain quite a bit about this bullshit, anyways, this should work for testing if we are being called on an EntityPlayerSP class "travel"...

Le sigh.

Reply:
you can try `((EntityPlayer) (Object) this).get class()`, I'm fairly certain that'll prevent the warning /j

Reply:
nope still throws in Intellij, either way the suppression is only for this line.

Reply:
So this works but it just throw a warning in the IDE?

Reply:
Yes it works, Mixin uses the compiled instructions to inject the values into the class. It works on my machine and what I read in Mixin's injection source. That's why you need to use the CaptureLocals to get anything you want. Redirect will redirect the call to your function and some other stuff.

I still want this tested in both SP and MP on multiple servers. It works on the ones I play on, but I want to make sure that EFly and other things are not broken.